client/components/jetpack/card/jetpack-product-card/display-price/time-frame.tsx

```diff
@@ -34,27 +45,78 @@ const TimeFrame: React.FC< TimeFrameProps > = ( { expiryDate, billingTerm } ) =>
 		};
 	}, [ billingTerm, translate ] );
 
-	return productExpiryDate ? (
-		<div>
-			<time
-				className="display-price__expiration-date"
-				dateTime={ productExpiryDate.format( 'YYYY-DD-YY' ) }
-			>
-				{ translate( 'expires %(date)s', {
-					args: {
-						date: productExpiryDate.format( 'L' ),
-					},
-				} ) }
-			</time>
-		</div>
-	) : (
-		<div>
-			<span className="display-price__billing-time-frame">
-				<span className="normal">{ billingTermText.normal }</span>
-				<span className="compact">{ billingTermText.compact }</span>
-			</span>
-		</div>
+	return (
+		<span className="display-price__billing-time-frame">
+			<span className="normal">{ billingTermText.normal }</span>
+			<span className="compact">{ billingTermText.compact }</span>
+		</span>
+	);
+};
+
+const ExpiringDateTimeFrame: React.FC< ExpiringDateTimeFrameProps > = ( { productExpiryDate } ) => {
+	const translate = useTranslate();
+	return (
+		<time
+			className="display-price__expiration-date"
+			dateTime={ productExpiryDate.format( 'YYYY-DD-YY' ) }
+		>
+			{ translate( 'expires %(date)s', {
+				args: {
+					date: productExpiryDate.format( 'L' ),
+				},
+			} ) }
+		</time>
 	);
 };
 
+const PartialDiscountTimeFrame: React.FC< PartialDiscountTimeFrameProps > = ( {
+	billingTerm,
+	discountedPriceDuration,
+} ) => {
+	const translate = useTranslate();
+
+	const BillingTermText = useMemo( () => {
+		if ( billingTerm === TERM_MONTHLY ) {
+			return discountedPriceDuration > 1
+				? translate( 'for the first %(months)d months, billed monthly', {
+						args: { months: discountedPriceDuration },
+				  } )
+				: translate( 'for the first month, billed monthly' );
+		}
+
+		return discountedPriceDuration > 1
+			? translate( 'for the first %(months)d months, billed yearly', {
+					args: { months: discountedPriceDuration },
+			  } )
+			: translate( 'for the first month, billed yearly' );
+	}, [ discountedPriceDuration, billingTerm, translate ] );
```

Comment:
Can we do something like this, to make it simpler? 
```suggestion
	const BillingTermText = useMemo( () => {
        const billingTermText = billingTerm === TERM_MONTHLY ? 'monthly' : 'yearly'
	    return discountedPriceDuration > 1
			? translate( 'for the first %(months)d months, billed %(term)s', {
					args: { months: discountedPriceDuration, term: billingTermText },
			   } )
		   : translate( 'for the first month, billed %(term)s',  { term: billingTermText });
	}, [ discountedPriceDuration, billingTerm, translate ] );
```

Reply:
That `yearly` and `monthly` needs to be translated. Thus it will need to be the way James did it.

Reply:
Oh yeah, that seems right. I didn't think about it. Then it is already looking great. LGTM